src/goInstallTools.ts

```diff
@@ -82,12 +84,22 @@ export async function installAllTools(updateExistingToolsOnly: boolean = false)
 		});
 }
 
+/**
+ * ToolAtVersion is a Tool with version annotation.
+ * Lack of version implies the latest version
+ */
+export interface ToolAtVersion extends Tool {
+	version?: SemVer;
```

Comment:
Why not add `version` as an optional property to `Tool` itself?

Reply:
Tool is used in various places, especially those in the allToolsInformation map, or other places worry me. Without properly cloning, the version information may be stored or not stored accidentally and misused. So, I decided to create an explicit versioned type.

Reply:
I am a little hesitant in introducing new interfaces unless absolutely required. Based on the changes in this PR, it looks like the code path for the prompt to update `gopls` is the only one that would use this `version` field. Do we foresee this feature to be extended to other tools? If not, then I would recommend having a separate function to prompt the update for `gopls` say `promptForUpdatingGopls` whose input parameter can be the version

Reply:
Nevermind, looks like the problem is going to be to plumb that information all through `installTools`

Reply:
I think that, in general, @hyangah was hoping to introduce the ability to pin tools to specific versions in the extension. That would fix the issue of tools breaking in certain versions, etc.